File: packages/nxuskit-go/solver_types.go
// Public CE solver placeholder types.
//
// Pro solver domain types are not shipped in public CE source or release bundles.
package nxuskit

type SolverStreamChunk struct {
	Error string `json:"error,omitempty"`
}

type SolverStreamResult struct {
	Chunks []SolverStreamChunk `json:"chunks,omitempty"`
}

File: packages/nxuskit/src/solver_types.rs
//! Public CE solver stream placeholder types.
//!
//! Pro solver domain types are not shipped in public CE source or release bundles.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct SolverStreamChunk {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct SolverStreamResult {
    #[serde(default)]
    pub chunks: Vec<SolverStreamChunk>,
}
